src/Common/Commands.Common/CustomAttributes/BreakingChangeAttributeHelper.cs

```diff
@@ -0,0 +1,151 @@
+
+    # New
+    # Sample of how the cmdlet should now be called
+    ```
+
+    Note: the above sections follow the template found in the link below:
+
+    https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes/breaking-change-template.md
+    -->";
+
+        public static void processCustomAttributesAtRuntime(Type type, List<String> boundParameterNames)
+        {
+            bool emitWarningOrError = true;
+
+            try
+            {
+                emitWarningOrError = Boolean.Parse(System.Environment.GetEnvironmentVariable(SUPPRESS_ERROR_OR_WARNING_MESSAGE_ENV_VARIABLE_NAME));
+            }
+            catch (Exception)
+            {
+                //no action
+            }
+
+            if (!emitWarningOrError)
+            {
+                //Do not process the attributes at runtime... The env variable to override the warning messages is set
+                return;
+            }
+
+            List<GenericBreakingChangeAttribute> attributes = getAllBreakingChangeAttributesInType(type, boundParameterNames);
+
+            if (attributes != null && attributes.Count > 0)
+            {
+                Console.WriteLine("Breaking changes in the cmdlet : " + Utilities.getNameFromCmdletType(type) + "\n");
+            }
+
+            foreach (GenericBreakingChangeAttribute attribute in attributes)
+            {
+                attribute.printCustomAttributeInfo(type, false);
+            }
+        }
+
+        public static List<String> getBreakingChangeMessagesForType(Type type)
+        {
+            List<String> messages = new List<string>();
+
+            //This is used as a migration guide, we need to process all properties/fields, moreover at this point of time we do not have a list of all the
+            //bound params anyways
+            foreach (GenericBreakingChangeAttribute attribute in getAllBreakingChangeAttributesInType(type, null))
+            {
+                messages.Add(attribute.getBreakingChangeTextFromAttribute(type, true));
+            }
+
+            return messages;
+        }
+
+        public static List<GenericBreakingChangeAttribute> getAllBreakingChangeAttributesInType(Type type, List<String> boundParameterNames)
+        {
+            List<GenericBreakingChangeAttribute> attributeList = new List<GenericBreakingChangeAttribute>();
+
+            attributeList.AddRange(type.GetCustomAttributes(typeof(GenericBreakingChangeAttribute), false).Cast<GenericBreakingChangeAttribute>());
+
+            foreach (MethodInfo m in type.GetRuntimeMethods())
+            {
+                attributeList.AddRange(m.GetCustomAttributes(typeof(GenericBreakingChangeAttribute), false).Cast<GenericBreakingChangeAttribute>());
+            }
+
+            //Only process the fields if the bound params have the field Name in them. If the bound params is null then
+            //we do no filtering
+            foreach (FieldInfo f in type.GetRuntimeFields())
```

Comment:
I don't think we actually need to worry about fields.  There are some extension methods defined here for getting cmdlet and parameter attributes that you may find useful: https://github.com/Azure/azure-powershell/blob/preview/tools/StaticAnalysis/HelpAnalyzer/ReflectionExtensions.cs

Reply:
fields are the ones that carry attributes for parameters...  it seems we need to com through them as well to gather custom attributes.